app/views/carts/_total_price.html.erb

```diff
@@ -1,6 +1,6 @@
 <p class="price-amount">SUBTotal :
   <span>
-    <%= number_to_currency(@cart.total_price) %>
+    <%= number_to_currency(@cart.decorate.total_price) %>
```

Comment:
Should call `decorate` in controller action. `number_to_currency` should also be put inside the decorator's method. It is better if we just call `@card.total_price` on view